cloud/libvirt/actuators/machine/utils/ignition.go

```diff
@@ -0,0 +1,188 @@
+package libvirt
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	libvirt "github.com/libvirt/libvirt-go"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+func CreateIgntion(pool, name, content string, client *Client) error {
+	log.Printf("[DEBUG] creating ignition file")
+	ignition := newIgnitionDef()
+
+	ignition.Name = name
+	ignition.PoolName = pool
+	ignition.Content = content
+
+	log.Printf("[INFO] ignition: %+v", ignition)
+
+	key, err := ignition.CreateAndUpload(client)
+	log.Printf("[INFO] Ignition ID: %s", key)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type defIgnition struct {
+	Name     string
+	PoolName string
+	Content  string
+}
+
+// Creates a new cloudinit with the defaults
+// the provider uses
+func newIgnitionDef() defIgnition {
+	return defIgnition{}
+}
+
+// Create a ISO file based on the contents of the CloudInit instance and
+// uploads it to the libVirt pool
+// Returns a string holding terraform's internal ID of this resource
+func (ign *defIgnition) CreateAndUpload(client *Client) (string, error) {
+	pool, err := client.libvirt.LookupStoragePoolByName(ign.PoolName)
+	if err != nil {
+		return "", fmt.Errorf("can't find storage pool '%s'", ign.PoolName)
+	}
+	defer pool.Free()
+
+	//client.poolMutexKV.Lock(ign.PoolName)
+	//defer client.poolMutexKV.Unlock(ign.PoolName)
+
+	// Refresh the pool of the volume so that libvirt knows it is
+	// not longer in use.
+	waitForSuccess("Error refreshing pool for volume", func() error {
+		return pool.Refresh(0)
```

Comment:
What happens if this never succeeds? This could run for long time in a CI job...

Reply:
as long as it is run as part of `_test.go` you can set timeout